sdks/python/src/opik/api_objects/dataset/dataset.py

```diff
@@ -27,6 +26,8 @@ def __init__(
         self._name = name
         self._description = description
         self._rest_client = rest_client
+        self._hash_to_id: Dict[str, str] = {}
+        self._id_to_hash: Dict[str, str] = {}
```

Comment:
Minor: no need to use two dictionaries to track this, you just need to set to track the duplicated items. Anyway, it just makes the logic a bit less straight forward and uses more space.